contract_function_selector.go

```diff
@@ -0,0 +1,224 @@
+package hedera
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/crypto/sha3"
+)
+
+type ContractFunctionSelector struct {
+	function   *string
+	params     string
+	paramTypes []solidity
+}
+
+type solidity struct {
+	ty    argument
+	array bool
+}
+
+type argument string
+
+const (
+	aBool     argument = "bool"
+	aString   argument = "string"
+	aInt32    argument = "int32"
+	aInt64    argument = "int64"
+	aInt256   argument = "int256"
+	aUint32   argument = "uint32"
+	aUint64   argument = "uint64"
+	aUint256  argument = "uint256"
+	aBytes    argument = "bytes"
+	aFunction argument = "function"
+	aAddress  argument = "address"
+)
+
+func NewContractFunctionSelector(function *string) ContractFunctionSelector {
```

Comment:
It might be better to have `""` mean `nil` for the alg here

--

As you can't just do `NewContractFunctionSelector(&"name")` in go ( need temp var )